### DISCUSSION TITLE
What is our aim?

### DISCUSSION THREAD

**Discussion by GForb** [pmsims-package/pmsims#12] [General]
How do we know when we are finished?

Sorting out sample size calculations for is a big task. We need to make sure we have an achievable, , aim (at least for phase 1 :)).

Currently I think we should try to:

1. Outline a general framework for simulation based sample size calculations in prediction models - ie. what people should use as performance metrics in their simulations. 
2. Develop a package which:
    - Implements a algorithm (using surrogate modelling) for finding the minimum sample size for a given data generating model, analysis method, and set of performance metrics
    - Contains a small number of preset data generating models, functions and performance metrics to make it easy for users who don't want to write their own.
    - Has enough examples/vignettes/etc that is is usable for people who aren't us
3. Publish package to CRAN and paper to a journal.

**Comment by GForb** [comment]
But what about clustered data...

I am interested in sample size for IPD meta-analysis where you have patients clustered in studies. Ewan is interested in repeated measures where you have observations clustered in patients. Felix has implemented simulations for cluster randomised trials.

When should we address sample sizes for clustered data?

It seems easy(ish - following Felix's work) and natural to extend to the situation where predictive performance is determined by number of clusters and observations per cluster and certainly useful.

I am wary about doing this straight off and wonder whether we should make sure we have a package out that can handle the situation in unclustered data (or when the number of clusters is fixed/set bu the user) for before we start addressing this.

**Reply by ewancarr** [reply]
Yes, I'm interested in clustered/longitudinal data.

I think we've discussed: (1) offering some 'default' options, e.g., `lm`, `glm`, `coxph`; (2) allowing the user to specify their own data generating function and model.

So, I was thinking this would be handled by (2). This does mean the user would need to specify how the model would be evaluated — i.e., provide a function that extracts the required metric from a fitted model.

But I'm not sure what other changes would be needed.

I suggest we focus on the defaults first, then move to (2) once the simpler cases work.

**Reply by GForb** [reply]
The complexity I see with clustered data is that there are two levels to the sample size. 1) number of clusters, 2) observations per cluster.

This required a different approach for surrogate modelling, (rather than just a different data generating function + model) as it is now a two dimensional optimization problem rather than one dimension. To find a single solution this requires some sort of cost function ie. how much does adding a cluster cost compared to adding a observation for an existing function. @flxzimmer solves this in his pre-print but the question is at what stage do we try implement for prediction. https://psyarxiv.com/r9w6t/

**Reply by flxzimmer** [reply]
I would also just do one-dimensional first and go from tere. If we went with surrogate modeling, in multiple dimensions, the biggest change would be the function to output the value of candidate search points would be more complex and can include a monetary cost function.

**Comment by ewancarr** [comment]
> How do we know when we are finished?

I agree with your list above. The only question is how far we go with "functions and performance metrics to make it easy for users who don't want to write their own". 

The easiest option would be to *not* allow user-generated data/models for the first release. We'd focus on the defaults (e.g., `lm`, `glm`, etc.) and comparisons to `pmsampsize`. 

We'd then add the user-generated functions in a subsequent release.

I'm wary of setting our initial aims too high.

**Reply by GForb** [reply]
"I'm wary of setting our initial aims too high." - 100% this

For me the user generated functions seems like the easy bit as we wouldn't have to come up with sensible defaults. But maybe much harder to explain how to use them/why it is useful.

If we go with default models I think we at least need to include penalised regression. If we are just looking at vanilla regression I'm not sure how much we add beyond pmsampsize. A sample size tool that take into account penalisation would be extremely useful

For performance metrics we can use standard options ie. calibration in the small/large c-statistic.

**Reply by flxzimmer** [reply]
I think not allowing user-generated data/models would not necessarily make it more manageable. I very much like the modular structure of the code with the different functions as placeholders for different aspects etc. Also the performance metrics I think could be just placeholders. We would fill all of these in a few examples and users could edit them / come up with new ones. I don't think there would be much pressure to come up with more than a few examples. We could collect ours in a vignette and offer to add user/reviewer-suggested ones. 

I also think penalised regression would be nice, possibly as the main example in the paper.

**Comment by flxzimmer** [comment]
Point (1) sounds a bit ambitious for me: "Outline a general framework for simulation based sample size calculations in prediction models - ie. what people should use as performance metrics in their simulations."

I can't think of an easy way to approach this right now, besides maybe gathering data from exisiting studies. 

I think that such guidelines should/need to evolve more naturally, I mean for example through meta analyses that show both the clinical impact of specific prediction models along with their numerical performance values. It would answer on which performance metrics succesful/unsuccesful prediction models actually differ and what desirable values are. Perhaps both of these might even differ with different predicted criteria.
